python/ray/tune/experimental/output.py

```diff
@@ -463,6 +464,31 @@ def __init__(self, verbosity: AirVerbosity):
         self._start_time = time.time()
         self._last_heartbeat_time = 0
 
+    def experiment_started(
```

Comment:
hey we should mark all the public APIs with @DeveloperAPI or something.
so it's very clear what functions one can override to completely customize the terminal output.
wdyt?

Reply:
Since it's currently experimental we don't need annotations just now (and should probably avoid them so we can continue to quickly iterate).

Before committing to the APIs, should we have a separate discussion on how and what users should be able to configure or overwrite? I don't think many people are going to write their own reporters tbh. It's more likely that they'll use configuration options to control the output, and the reporter interface is for us to implement new handlers/features if needed.